Importamos librerias y base de datos a trabajar

In [2]:
import pandas as pd

events = pd.read_csv('events.csv')

Creamos la tabla pivote para poder trabajar en ella

In [3]:
users = events.pivot_table(
    index='uid', 
    columns='event_name', 
    values='event_time',
    aggfunc='min')

Confirmamos paso por paso de cada parte del embudo

In [4]:
step_1 = ~users['pageview'].isna() 
step_2 = step_1 & (users['add_to_cart'] > users['pageview']) 
step_3 = step_2 & (users['checkout'] > users['add_to_cart'])
step_4 = step_3 & (users['payment'] > users['checkout'])

Contamos los ususarios que completaron cada paso

In [5]:
n_pageview = users[step_1].shape[0] 
n_add_to_cart = users[step_2].shape[0] 
n_checkout = users[step_3].shape[0]
n_payment = users[step_4].shape[0]

Imprimimos los resultados

In [6]:
print('Visitantes:', n_pageview)
print('Se agregó un producto al carrito:', n_add_to_cart)
print('Pago iniciado:', n_checkout)
print('Pagado:', n_payment)

Visitantes: 8831
Se agregó un producto al carrito: 2584
Pago iniciado: 354
Pagado: 28


## Conclusión 
Fueron 8831 usuarios los que visitaron la página de compras ya sea través de un banner, correo electrónico o recomendación,
- de los cuales 2584 les interesó algún producto y lo agregaron a su carrito,
- 354 procedieron a la página de pago 
- y solo 28 de ellos realizaron la compra, (debe de ser porque al momento del pago se muestra el costo del envío y muchos ahí se quedan)